src/XPurchase/VirtoCommerce.XPurchase/CartAggregate.cs

```diff
@@ -819,16 +815,12 @@ public void SetLineItemTierPrice(ProductPrice productPrice, int quantity, LineIt
         {
             if (productPrice == null) return;
 
-            var salePrice = productPrice.GetTierPrice(quantity).Price;
-            if (salePrice != 0)
+            var TierPrice = productPrice.GetTierPrice(quantity).Price;
+            if (TierPrice.Amount > 0)
             {
-                lineItem.SalePrice = salePrice.Amount;
+                lineItem.SalePrice = TierPrice.Amount;
+                lineItem.ListPrice = lineItem.SalePrice;

```

Comment:
Try to change on lineItem.ListPrice = tierPrice.Price;

Reply:
tierPrice does not contain Price